config/crds/kudo.dev_instances.yaml

```diff
@@ -3,44 +3,161 @@
+                    For example, if the object reference is to a container within
+                    a pod, this would take on a value like: "spec.containers{name}"
+                    (where "name" refers to the name of the container that triggered
+                    the event) or if no container name is specified "spec.containers[2]"
+                    (container with index 2 in this pod). This syntax is chosen only
+                    to have some well-defined way of referencing a part of an object.
+                    TODO: this design is not final and this field is subject to change
+                    in the future.'
+                  type: string
+                kind:
+                  description: 'Kind of the referent. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+                  type: string
+                name:
+                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                  type: string
+                namespace:
+                  description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                  type: string
+                resourceVersion:
+                  description: 'Specific resourceVersion to which this reference is
+                    made, if any. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency'
+                  type: string
+                uid:
+                  description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
+                  type: string
               type: object
             parameters:
+              additionalProperties:
+                type: string
               type: object
           type: object
         status:
+          description: InstanceStatus defines the observed state of Instance
           properties:
             aggregatedStatus:
+              description: AggregatedStatus is overview of an instance status derived
+                from the plan status
+              properties:
+                activePlanName:
+                  type: string
+                status:
+                  description: ExecutionStatus captures the state of the rollout.
+                  type: string
               type: object
             planStatus:
+              additionalProperties:
+                description: "PlanStatus is representing status of a plan \n These
+                  are valid states and transitions \n                       | Never
+                  executed |                               |                               v
+                  |    Error    |<------>|    Pending     |       ^                       |
```

Comment:
eh, this does not look good

Reply:
I agree, although I'm assuming nobody will be treating this file as documentation, all the more these are our "internal" CRDs, so I don't know how much energy it makes sense to invest to improve how this ASCII-art looks here.

Reply:
I agree about internal documentation. But this still looks awkward. Should we maybe separate this diagram from the `PlanStatus` doc so it simply doesn't appear here? Wdyt @porridge ?

Reply:
@zen-dog yes, that was one option I had in mind.
The other would be to leave it as is, and then look at some tools which would take the CRDs as input and generate nice user-friendly documentation. In that case this ASCII-art might survive and actually be useful? Perhaps this is something worth looking at when we work on docs next week?